### ISSUE TITLE
MANIFEST.MF from project is not added to the proguarded jar

### ISSUE THREAD

**Issue by soc** [sbt/sbt-proguard#2]
For comparison, this works flawlessly without any additional work in sbt-assembly.

I was able to exclude random MANIFEST.MF files from jar dependencies with

```
ProguardKeys.inputFilter in Proguard := { file =>
  file.name match {
    case "scala-library-2.11.0-M8.jar"          => Some("!META-INF/**")
    case "regextractor-core_2.11.0-M8-0.1.jar"  => Some("!META-INF/**")
    case "regextractor-util_2.11.0-M8-0.1.jar"  => Some("!META-INF/**")
    case "rxjava-core-0.16.2-SNAPSHOT.jar"      => Some("!META-INF/**")
    case "rxjava-scala-0.16.2-SNAPSHOT.jar"     => Some("!META-INF/**")
    case "seaglasslookandfeel-0.2.jar"          => Some("!META-INF/**")
    case _ => None
  }
}
```

but I still can't figure out how to add the MANIFEST.MF file of my own project.

sbt package and sbt-assembly seem to derive that information automatically from the build script, but even trying to add the file manually with

```
mappings in (Compile, packageBin) <+= baseDirectory map { base =>
   (base / "MANIFEST.MF") -> "META-INF/MANIFEST.MF"
}
```

doesn't have any effect with sbt-proguard.

**Comment by AlexLourenco** [comment]
While this might no longer be relevant to those who have posted in this issue (given how old this thread is), here is the solution in case newcomers experience the same problem:

As pvlugter suggested, setting up exportJars := true will have SBT generating the jar file, which is then extracted alongside its dependencies into the merge folder. Assuming we are only interested in keeping the MANIFEST.MF of the current project, one can simply keep the first found manifest (which will always be the one from project.jar) - first("META-INF/MANIFEST.MF"), and discard everything else.

Now that the manifest is being kept the following step is to override the default inputFilter by either providing a custom filter like in soc's example, or if no custom logic is required, simply returning none.

Here's an example of the configuration; https://gist.github.com/AlexLourenco/8421ddf982547d5cc665

**Comment by soc** [comment]
@AlexLourenco Well, I'd still like to use it. :-) On a note, I think ProGuard 5.1 should work now, because the corresponding Scala issue has been fixed in 2.11.5.
